### ISSUE TITLE
DNSimple v2

### ISSUE THREAD

**Issue by nanliu** [PalmStoneGames/kube-cert-manager#56]
Looks like the current glide dependency uses the old DNSimple v1 API written by weppos. The lego repo have been updated to [DNSimple official go library](https://github.com/xenolf/lego/pull/363).

When I tried master, it seems to have issues with the token (verified working separately)
```
2017/04/19 00:23:21 Error while processing certificate event: Error while obtaining certificate for new domain [omitted]: Error presenting token: DNSimple API call failed: GET https://api.dnsimple.com/v1/domains: 401 Authentication failed
```

I think we can update glide, and replace DNSIMPLE_API_KEY with DNSIMPLE_OAUTH_TOKEN, however the requirement for DNSIMPLE_BASE_URL might be more annoying since it will require multiple kube-cert-manager per domain.

**Comment by luna-duclos** [comment]
Would updating the package actually be useful to DNSSIMPLE users ? It seems to me like xenolf/lego probably needs a way to pass in the baseURL and we need to support that

**Comment by nanliu** [comment]
Yes we should update to v2, because DNSimple is deprecating their v1 API, and it will be no longer available as of October 1st, 2017. We can pass the baseURL via environment variable.

**Comment by luna-duclos** [comment]
Closed by #57, thank you!

**Comment by nanliu** [comment]
PR #50 clobbered over #57, we are back to DNSimple v1.
